### PR TITLE
Add service account as project member

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
With this PR it's possible to let the `gardener-controller-manager` manage the rbac for the temporary access service account by adding it to the project as member with the corresponding roles.
The specified target credential needs the permission to manage the roles of the project.
Of course and as usual, the user that creates the terminal resource needs the permission to read the target credential

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
It's now possible to let the `gardener-controller-manager` manage the rbac for the temporary access service account by adding it to the project as member with the corresponding roles.
```

### CODE REVIEW

File: config/samples/terminal-garden.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        namespace: garden-coretmp
+    kubeconfigContextNamespace: garden-coretmp
+    namespace: garden-coretmp
+    authorization:
+      projectMemberships:
+        - projectName: core-tmp
+          roles:
+            - admin
+    apiServer:
+      server: https://kubernetes.default.svc
```
Comment (petersutter) [inline config/samples/terminal-garden.yaml:31]: maybe move under
kubeconfig:
  contextNamespace: garden-coretmp
  cluster/apiServer...

File: main.go [hunk 2]
```diff
@@ -54,6 +55,7 @@ func init() {
 	_ = clientgoscheme.AddToScheme(scheme)
 
 	_ = v1alpha1.AddToScheme(scheme)
+	_= gardencorev1beta1.AddToScheme(scheme)
```
Comment (petersutter) [inline main.go:58]:
```suggestion
	_ = gardencorev1beta1.AddToScheme(scheme)
```

File: go.mod [hunk 3]
```diff
@@ -3,32 +3,26 @@ module github.com/gardener/terminal-controller-manager
 go 1.15
 
 require (
-	cloud.google.com/go v0.46.3 // indirect
-	github.com/beorn7/perks v1.0.1 // indirect
-	github.com/fsnotify/fsnotify v1.4.9 // indirect
+	github.com/gardener/gardener v1.11.0
```
Comment (petersutter) [inline go.mod:6]: 1.12

File: main.go (hunk 2 above)
Comment (petersutter) [reply main.go]: ✅

File: go.mod (hunk 3 above)
Comment (petersutter) [reply go.mod]: ✅

**Review by grolu — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**